frontend/src/advising/Templates/NewTemplatesPage.tsx

```diff
@@ -0,0 +1,148 @@
+import { TextField, FormControl } from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import React from "react";
+import { useState } from "react";
+import { useSelector } from "react-redux";
+import styled from "styled-components";
+import { getMajorsFromState, getPlansFromState } from "../../state";
+import { AppState } from "../../state/reducers/state";
+import { planToString } from "../../utils";
+import { WhiteColorButton, ColorButton } from "../GenericAdvisingTemplate";
+import { TemplatePageState } from "./Templates";
+
+const Container = styled.div`
+  margin-left: 30px;
+  margin-right: 30px;
+  margin-top: 50px;
+  font-family: Roboto;
+  font-style: normal;
+  font-weight: normal;
+  overflow: hidden;
+`;
+
+const NewTemplatesPageContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 40px;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const InputContainer = styled.div`
+  width: 326px;
+`;
+
+interface TemplatesPageProps {
+  readonly setPageState: (pageState: TemplatePageState) => void;
+}
+
+interface DropdownProps {
+  readonly label: string;
+  readonly options: Array<string>;
+  readonly value?: string;
+  readonly setValue: (value?: string) => void;
+}
+
+interface NameFieldProps {
+  readonly name: string;
+  readonly setTemplateName: (name: string) => void;
+}
+
+export const NewTemplatesPage: React.FC<TemplatesPageProps> = ({
+  setPageState,
+}) => {
+  const [name, setName] = useState("");
+  const [major, setMajor] = useState<string | undefined>(undefined);
+  const [catalogYear, setCatalogYear] = useState<string | undefined>(undefined);
+  const [coopCycle, setCoopCycle] = useState<string | undefined>(undefined);
+
+  const majors = useSelector((state: AppState) => getMajorsFromState(state));
+  const catalogYears = [
+    ...Array.from(new Set(majors.map(maj => maj.yearVersion.toString()))),
+  ];
+  const coopCycles = useSelector((state: AppState) => getPlansFromState(state));
+  const buttonSize = 90;
+  const disabled = !(name && major && catalogYear && coopCycle);
+
+  return (
+    <NewTemplatesPageContainer>
+      <Container style={{ fontSize: "24px" }}>
+        Let's create a template!
+      </Container>
+      <InputContainer>
+        <NameField name={name} setTemplateName={setName} />
+      </InputContainer>
+      <Dropdown
+        label="Major"
+        options={majors.map(maj => maj.name)}
+        value={major}
+        setValue={setMajor}
+      />
+      <Dropdown
+        label="Catalog year"
+        options={catalogYears}
+        value={catalogYear}
+        setValue={setCatalogYear}
+      />
+      {major && (
+        <Dropdown
+          label="Co-op cycle"
+          options={coopCycles[major!].map(p => planToString(p))}
+          value={coopCycle}
+          setValue={setCoopCycle}
+        ></Dropdown>
+      )}
```

Comment:
Since showing co-op cycles requires a major to be selected, we hide the input if there is no major